lib/jekyll-admin/server/collection.rb

```diff
@@ -29,7 +29,9 @@ class Server < Sinatra::Base
         end
 
         write_file(write_path, document_body)
-        updated_document = collection.docs.find { |d| d.path == write_path }
+        updated_document = Jekyll::Document.new(write_path, {
+          :collection => collection, :site => site,
+        }).tap(&:read)
         render_404 if updated_document.nil?
```

Comment:
Isn't `document` overwritten by `Jekyll::Document.new` ? So `ensure_document` uses newly created document object instead of the result object of `document` method.